libraries/Microsoft.Bot.Builder.LanguageGeneration/TemplateException.cs

```diff
@@ -25,6 +29,8 @@ public TemplateException(string message, IList<Diagnostic> diagnostics)
         /// <value>
         /// Diagnostics.
         /// </value>
+#pragma warning disable CA2227 // Collection properties should be read only (we can't remove the setter without breaking binary compat)
```

Comment:
Please double check whether we can forego binary compat, depending on the larger decision of whether we'll continue to guarantee binary level compat.

Reply:
Per discussion Discussed this with @axelsrz over email, removing the setter would break source compat so I am leaving it as is.